Infrastructure: fix MacOS run time crashes due to missing libzstd

SlySven opened the pull request:
#### Brief overview of PR changes/additions
This tries to improve things by ensuring the library is now explicitly listed as a package to have installed. Whether that is sufficient to fix the problem needs to be tested. If it does not solve the problem we may need to look to see how we specify what libraries/frameworks are required in the bundled application that we provide on MacOS.

#### Motivation for adding to Mudlet
We have recently started to see crashes because MacOS builds do not appear to be finding `libzstd.1.dylib` to dynamically link to at run-time as needed by `libzip-5.dylib`, e.g. here:
https://github.com/Mudlet/Mudlet/pull/6982#issuecomment-1752202617

#### Other info (issues closed, discussion etc)
 I can see that (with the `dmg2img` Linux utility) the library does seem to be absent from our product currently (the recent PR test build that is mentioned about) but **it is present, (as the `mudlet.app/Contents/Frameworks/libzstd.1.dylib` file) in our most recent *Release* version...**

SlySven committed:
Infrastructure: fix MacOS run time crashes due to missing libzstd

We have recently started to see crashes because MacOS builds do not appear
to be finding `libzstd.1.dylib` to dynamically link to at run-time as
needed by `libzip-5.dylib`, e.g. here:
https://github.com/Mudlet/Mudlet/pull/6982#issuecomment-1752202617

This tries to improve things by ensuring the library is now explicitly
listed as a package to have installed. Whether that is sufficient to fix
the problem needs to be tested. If it does not solve the problem we may
need to look to see how we specify what libraries/frameworks are required
in the bundled application that we provide on MacOS. I can see that (with
the `dmg2img` Linux utility) the library does seem to be absent from our
product currently (the recent PR test build that is mentioned about) but
**it is present, (as the `mudlet.app/Contents/Frameworks/libzstd.1.dylib`
file) in our most recent *Release* version...**

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Humm, https://make.mudlet.org/snapshots/50197b/Mudlet-4.17.2-testing-pr6998-544c4f6e.dmg did NOT contain the wanted lib `libzstd.1.dylib` in `Mudlet-4.17.2-testing-pr6998-544c4f6e.app/Contents/MacOS/Frameworks/` . So now we need to look deeper...

SlySven committed:
Revise: increase minimum CMake version required & force some lib re-installs

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: undo change of minimum CMake version

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: add some MacOS CB specific debug output [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: another attempt to get some MacOS Homebrew environment details [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: run brew checkup steps in a separate job [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Infrastructure_fixMacOSRunTimeCrashesDueToMissing_libzstd

SlySven committed:
Revise: fix buggy indentation [skip appveyor]

YAML is indent sensitive - and I had not noticed I had messed it up!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Infrastructure_fixMacOSRunTimeCrashesDueToMissing_libzstd

This will update the node brew from version 18 to 20 which is one of the things
that "brew doctor" reported from a previous commit.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: re-enable brew auto-updates [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise more experimenting [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: specifically and explicitly update brews [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: change from brew update to brew upgrade (recipe) [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: be more specific about the Ruby version

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use brew upgrade instead of brew update

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: revert back, just try to find the problem items [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: try a different invocation of mdfind [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into Infrastructure_fixMacOSRunTimeCrashesDueToMissing_libzstd

vadi2 commented:
@SlySven if you add https://github.com/marketplace/actions/debugging-with-tmate to the workflow, you can then ssh into the runner machine - I find that speeds debugging up significantly.

SlySven committed:
Debug: add a call to tmate to allow for remote login to CB [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: remove a duplicate "uses" line from yaml file in last commit [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: remove timeout for SSH session into MacOS CI builds [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Test: use alternate installer branch which copies over missing libzstd

This is only a test build to see if the concept (copying the files
directly) is a valid solution.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: change repo for installer & reinsert SSH login to CI process [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Undo: remove SSH login to MacOS CI process

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
No luck using Mudlet-4.17.2-testing-pr6998-33b92c4c unfortunately.

SlySven commented:
How does it behave - does it crash in *exactly* the same way as https://github.com/Mudlet/Mudlet/pull/6982#issuecomment-1752202617 - or is the error message different?

SlySven committed:
Revise: refactor CI/travis.osx.after_success.sh

This is so that there is less branching depending on whether it is a PTB or
release build - only one test rather than lots. This is so that it will be
easier to split the file up into two pieces - one for assembling the
package and one for signing it and uploading to our site.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: split MacOS CI/travis.osx.after_success.sh file into two parts

This is so that we can (temporarily) stop the packaging before it gets
signed and then check and revise the package.

Also some bash variables are renamed to SCREAMING_SNAKE_CASE to make them
easier to pick out - with some single use ones replaced with their
definitions.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: re-add SSH login for MacOS builds [skip appveyor]

This now happens after the package has been assembled but before it is
signed off - so that we can investigate/debug/fix the current problem with
the libzip and its now dependency on libzstd library.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: OS specific stage error in build-mudlet.yml file [skip appveyor]

The particular stage should have been made Linux only in previous commit.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: remove unneeded and thus broken git clone in MacOS build [skip appveyor]

It broke things because it failed as it had already been done in an earlier
stage.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: make a trivial change to trigger a complete PR CI re-run

Needed to generate a new set of test builds.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
@vadi2 try https://make.mudlet.org/snapshots/b09061/Mudlet-4.17.2-testing-pr6998-4a23e6af.dmg and see if that works...

SlySven commented:
As such the changes I've made in the installer repository - that this PR made use of: https://github.com/Mudlet/installers/pull/111 before that was merged now means this PR is not now needed - so I will close it.

Some aspects (SCREAMING_SNAKE_CASING some variables in some scripts, reorganising things so that only one "if" to divide things up between PR/Testing/PTB and Release builds) might be worth doing but should be split out into separate, smaller commits in new PRs.

SlySven commented:
Reopening because the (patch) version of the `xz` library and the `liblzma`  dependency (the latter which `libzip` requires) has changed from **5.4.4** to **5.4.*5*** and this was enough to to break things again. I need this PR's code temporarily again as it has the ability to pause the MacOS build at the point where investigations and fixes can be done.

SlySven committed:
Merge branch 'development' into Infrastructure_fixMacOSRunTimeCrashesDueToMissing_libzstd

SlySven committed:
Fix: change installer repository for liblzma library version [skip appveyor]

Not for merging into `development` just for testing/developing installer
repository script.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Closing again as it has done it's work...
